source/mod_dist.f90

```diff
@@ -20,45 +23,172 @@
+  integer, parameter :: dist_fmtE           = 22 ! Particle energy
+  integer, parameter :: dist_fmtP           = 23 ! Particle momentum
+  integer, parameter :: dist_fmtDEE0        = 24 ! Relative particle energy (to reference particle)
+  integer, parameter :: dist_fmtDPP0        = 25 ! Relative particle momentum (to reference particle)
+  integer, parameter :: dist_fmtPT          = 26 ! Delta enery over reference momentum (Pt)
+
+  ! Normalised Coordinates
+  integer, parameter :: dist_fmtXN          = 31 ! Normalised horizontal position
+  integer, parameter :: dist_fmtYN          = 32 ! Normalised vertical position
+  integer, parameter :: dist_fmtZN          = 33 ! Normalised longitudinal position
+  integer, parameter :: dist_fmtPXN         = 34 ! Normalised horizontal momentum
+  integer, parameter :: dist_fmtPYN         = 35 ! Normalised vertical momentum
+  integer, parameter :: dist_fmtPZN         = 36 ! Normalised longitudinal momentum
+
+  integer, parameter :: dist_fmtJX          = 41 ! Horizontal action
+  integer, parameter :: dist_fmtJY          = 42 ! Vertical action
+  integer, parameter :: dist_fmtJZ          = 43 ! Longitudinal action
+  integer, parameter :: dist_fmtPhiX        = 44 ! Horizontal action angle
+  integer, parameter :: dist_fmtPhiY        = 45 ! Vertical action angle
+  integer, parameter :: dist_fmtPhiZ        = 46 ! Longitudinal action angle
+
+  ! Ion Columns
+  integer, parameter :: dist_fmtMASS        = 51 ! Particle mass
+  integer, parameter :: dist_fmtCHARGE      = 52 ! Particle Charge
+  integer, parameter :: dist_fmtIonA        = 53 ! Ion atomic mass
+  integer, parameter :: dist_fmtIonZ        = 54 ! Ion atomic number
+  integer, parameter :: dist_fmtPDGID       = 55 ! Particle PDG ID
+
+  ! Flags for columns we've set that we need to track for later checks
+  logical, private, save :: dist_readMass   = .false.
+  logical, private, save :: dist_readIonZ   = .false.
+  logical, private, save :: dist_readIonA   = .false.
+  logical, private, save :: dist_readCharge = .false.
+  logical, private, save :: dist_readPDGID  = .false.
+  logical, private, save :: dist_norm4D     = .false.
+  logical, private, save :: dist_norm6D     = .false.
+
+  ! Arrays to send to DISTlib
+  real(kind=fPrec), allocatable, private, save :: dist_partCol1(:)    ! Ends up in array xv1
+  real(kind=fPrec), allocatable, private, save :: dist_partCol2(:)    ! Ends up in array yv1
+  real(kind=fPrec), allocatable, private, save :: dist_partCol3(:)    ! Ends up in array xv2
+  real(kind=fPrec), allocatable, private, save :: dist_partCol4(:)    ! Ends up in array yv2
+  real(kind=fPrec), allocatable, private, save :: dist_partCol5(:)    ! Ends up in array sigmv
+  real(kind=fPrec), allocatable, private, save :: dist_partCol6(:)    ! Ends up in array dpsv, evj, and ejfv
+  integer,                       private, save :: dist_partFmt(6) = 0 ! The format used for each column
```

Comment:
in order to be synched with `dist_postParticles`, I think you should initialise `dist_partFmr(6)=dist_fmtNONE`

Reply:
That is the intention, yes. Still not finished the rewriting after the latest round of changes.